.github/workflows/sonarqube-scan-pr.yml

```diff
@@ -0,0 +1,25 @@
+on:
+  # Warning: using the pull_request_target event without the cautionary measures may allow
+  # unauthorized GitHub users to open a “pwn request” and exfiltrate secrets.
+  pull_request_target:
+    types: [opened, synchronize, reopened, labeled]
+
+name: SonarCloud Code Scan
+jobs:
+  # For monorepos, create one job per project and specify the directory
+  sonarqube-src:
+    environment: ${{ github.event_name == 'pull_request_target' &&
+      github.event.pull_request.head.repo.full_name != github.repository && 'sonar-pr' || 'none' }}
```

Comment:
Can you help me understand this environment config? The event will always be pull_request_target, so I think the first condition can be removed (unless pull_request events are a subset of pull_request_target events--but I don't think that's true).

So if it's in Sage-Bionetworks, it will be 'sonar-pr' and otherwise it will be 'none'? What does this do?

Reply:
This is to ensure the workflow runs on PRs initiated from a fork, not within the main repo. If so, it will run in the `sonar-pr` environment, which contains the `SONAR_TOKEN` variable and is configured so our team members can approve workflows initiated by forks. Otherwise, if no environment is selected, this workflow should fail because forks can't access secrets.